### PR TITLE
update quoteservice to opentelemetry-php beta

### PR DESCRIPTION
**Pull request by brettmc**
# Changes

- update to opentelemetry-php beta versions
- import only required opentelemetry-php libraries
- use php auto-instrumentation (remove most manual instrumentation, leaving enough to illustrate that auto+manual can work together)
- bump php to 8.2, the most recent version supported by opentelemetry-php

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [x] Appropriate documentation updates in the [docs](../docs/) folder

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -435,14 +435,15 @@ services:
     ports:
       - "${QUOTE_SERVICE_PORT}"
     environment:
-      # OTEL_EXPORTER_OTLP_TRACES_ENDPOINT # Not working for PHP
+      - OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=http://otelcol:4318/v1/traces
+      #- OTEL_EXPORTER_OTLP_ENDPOINT=http://otelcol:4318
```
Comment (puckpuck) [inline docker-compose.yml:439]: please delete this line completely if not needed

File: docker-compose.yml [hunk 2]
```diff
@@ -435,14 +435,15 @@ services:
...
+      - OTEL_PHP_AUTOLOAD_ENABLED=true
       - QUOTE_SERVICE_PORT
       - OTEL_SERVICE_NAME=quoteservice
-      - OTEL_EXPORTER_OTLP_ENDPOINT=http://otelcol:4318
       - OTEL_TRACES_SAMPLER=parentbased_always_on
       - OTEL_TRACES_EXPORTER=otlp
       - OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=http/protobuf
-      - OTEL_PHP_TRACES_PROCESSOR=simple
+      - OTEL_PHP_TRACES_PROCESSOR=batch
```
Comment (puckpuck) [inline docker-compose.yml:446]: is `batch` the default now? If so we can remove this environment variable

**Review by puckpuck — CHANGES_REQUESTED** [review]
Please update the [docs/servcies/quoteservice.md](https://github.com/open-telemetry/opentelemetry-demo/blob/main/docs/services/quoteservice.md) docs and create a changelog entry.

File: docker-compose.yml (hunk 2 above)
Comment (brettmc) [reply docker-compose.yml]: yes, batch is default. I have also removed other env vars that were using default values

**Review by puckpuck — APPROVED** [review]
Thank you for this!

**Review by julianocosta89 — CHANGES_REQUESTED** [review]
This looks great @brettmc, thanks for that!

I've tested it locally and it seems that we have some issues though.
After building this `quoteservice` I've started getting the following:

```
quoteservice             | [Mon Dec 19 06:26:48.124913 2022] [php:notice] [pid 19] [client 172.18.0.16:57740] Slim Application Error\nType: Error\nCode: 0\nMessage: Class &quot;OpenTelemetry\\API\\Trace\\Span&quot; not found\nFile: /var/www/app/routes.php\nLine: 41\nTrace: #0 [internal function]: {closure}(Object(Slim\\Psr7\\Request), Object(Slim\\Psr7\\Response))\n#1 /var/www/vendor/php-di/invoker/src/Invoker.php(74): call_user_func_array(Object(Closure), Array)\n#2 /var/www/vendor/php-di/slim-bridge/src/ControllerInvoker.php(47): Invoker\\Invoker->call(Object(Closure), Array)\n#3 /var/www/vendor/slim/slim/Slim/Routing/Route.php(384): DI\\Bridge\\Slim\\ControllerInvoker->__invoke(Object(Closure), Object(Slim\\Psr7\\Request), Object(Slim\\Psr7\\Response), Array)\n#4 /var/www/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Slim\\Routing\\Route->handle(Object(Slim\\Psr7\\Request))\n#5 /var/www/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Slim\\MiddlewareDispatcher->handle(Object(Slim\\Psr7\\Request))\n#6 /var/www/vendor/slim/slim/Slim/Routing/Route.php(341): Slim\\MiddlewareDispatcher->handle(Object(Slim\\Psr7\\Request))\n#7 /var/www/vendor/slim/slim/Slim/Routing/RouteRunner.php(84): Slim\\Routing\\Route->run(Object(Slim\\Psr7\\Request))\n#8 /var/www/vendor/slim/slim/Slim/Middleware/RoutingMiddleware.php(59): Slim\\Routing\\RouteRunner->handle(Object(Slim\\Psr7\\Request))\n#9 /var/www/vendor/slim/slim/Slim/MiddlewareDispatcher.php(147): Slim\\Middleware\\RoutingMiddleware->process(Object(Slim\\Psr7\\Request), Object(Slim\\Routing\\RouteRunner))\n#10 /var/www/vendor/slim/slim/Slim/Middleware/BodyParsingMiddleware.php(68): Psr\\Http\\Server\\RequestHandlerInterface@anonymous->handle(Object(Slim\\Psr7\\Request))\n#11 /var/www/vendor/slim/slim/Slim/MiddlewareDispatcher.php(147): Slim\\Middleware\\BodyParsingMiddleware->process(Object(Slim\\Psr7\\Request), Object(Psr\\Http\\Server\\RequestHandlerInterface@anonymous))\n#12 /var/www/vendor/slim/slim/Slim/Middleware/ErrorMiddleware.php(107): Psr\\Http\\Server\\RequestHandlerInterface@anonymous->handle(Object(Slim\\Psr7\\Request))\n#13 /var/www/vendor/slim/slim/Slim/MiddlewareDispatcher.php(147): Slim\\Middleware\\ErrorMiddleware->process(Object(Slim\\Psr7\\Request), Object(Psr\\Http\\Server\\RequestHandlerInterface@anonymous))\n#14 /var/www/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Psr\\Http\\Server\\RequestHandlerInterface@anonymous->handle(Object(Slim\\Psr7\\Request))\n#15 /var/www/vendor/slim/slim/Slim/App.php(215): Slim\\MiddlewareDispatcher->handle(Object(Slim\\Psr7\\Request))\n#16 /var/www/vendor/slim/slim/Slim/App.php(199): Slim\\App->handle(Object(Slim\\Psr7\\Request))\n#17 /var/www/public/index.php(56): Slim\\App->run()\n#18 {main}
quoteservice             | 172.18.0.16 - - [19/Dec/2022:06:26:48 +0000] "POST /getquote HTTP/1.1" 500 3612 "-" "-"
```

Could you take a look?

**Review by julianocosta89 — APPROVED** [review]
I've built it in my Ubuntu and it worked.
Not sure what happened on Windows, but it LGTM!

**Comment by julianocosta89** [comment]
Just adding a screenshot here for visibility:
Screenshot from 2022-12-19 20-07-56

With this change `quoteservice` produces 14 spans per request.

**Review by fatsheep9146 — APPROVED** [review]

**Review by bobstrecansky — APPROVED** [review]

**Comment by cartersocha** [comment]
Lgtm. Merging. 

Thanks @brettmc for the contribution! 🦞

**Comment by cartersocha** [comment]
And @bobstrecansky for taking time to help with the review 😎

**Comment by puckpuck** [comment]
took me a while to test this one out, but I also see the same as @julianocosta89 did.  The quoteservice now produces several auto-instrumented spans, well beyond what should be expected.

@brettmc can you please take a look at this? I'm going to open an issue as well.
Screen Shot 2022-12-24 at 12 40 24 AM

**Pull request merged**